shared/node.go

```diff
@@ -23,28 +23,28 @@ type Node struct {
 	Address cryptoPocket.Address
 }
 
-func Create(cfg *genesis.Config, genesis *genesis.GenesisState) (n *Node, err error) {
-	persistenceMod, err := persistence.Create(cfg, genesis)
+func Create(cfg, genesis map[string]json.RawMessage) (n *Node, err error) {
```

Comment:
I'm not a fan of moving from something with semantic meaning to something like`map[string]json.RawMessage` everywhere like this.

I think we could (should?) create a `GenesisStruct` and `Config` that has a `popualte` or `hydrate` function, that has well defined elements instead of string keys

```
type GenesisStruct struct {
   P2P:
   Consensus:
   Persistence:
   Utility:
}
```

The `populate` function would:

1. Panic if the key is not available (right now we may or may not be trying to access non-avalble keys)
2. This will centralize the need to call `InitGenesis` and `InitiConfig` in one file and reduce the code bloat everywhere else where we just want to focus on the business logic of each module.

Reply:
While this is being used in production for all Tendermint projects today, I too think the design can be improved. 

I think this is plenty sufficient for the current implementation, please open a follow up issue with your suggestion

Reply:
I don't think we should try and follow in `Tendermint's` path because the reason we're building v1 from scratch is so we can do a better job from the ground up.

I feel quite strongly about making this change and believe it is a blocker from moving this PR forward. The fact that we had a semantic definition for the protobuf and are now moving back to a `RawJson` feels like a step back in terms of the quality of the codebase even though we are adding more functionality.

I believe that Given that @adshmh (a newcomer to the repository) had the same impression shows we need to focus on it for now.

Let's figure out a good design for this together next week before continuing the review of this PR.